lib/presentation/mixins/connect_page_mixin.dart

```diff
@@ -286,4 +301,137 @@ mixin ConnectPageMixin {
       '${html.window.location.href.getBaseUrlBeforeHash()}#/${route ?? 'rooms'}',
     );
   }
+
+  Future<Client?> getClientExisted({required String homeServer}) async {
+    try {
+      final clients = await ClientManager.getClients();
+      Logs().d(
+        'ConnectPageMixin::getClientExisted:Clients = ${clients.map((client) => client.homeserver).toString()}',
+      );
+      final clientExisted = clients.firstWhereOrNull(
+        (client) => client.homeserver?.toString().contains(homeServer) == true,
+      );
+      Logs().d(
+        'ConnectPageMixin::getClientExisted: $clientExisted',
+      );
+      if (clientExisted != null &&
+          !AppConfig.supportMultipleAccountsInTheSameHomeserver) {
+        return clientExisted;
+      } else {
+        return null;
+      }
+    } catch (e) {
+      Logs().e(
+        'ConnectPageMixin::getClientExisted: Exception: $e',
+      );
+      return null;
+    }
+  }
+
+  Future<void> logoutAction({required MatrixState matrix}) async {
+    Logs().d('ConnectPageMixin::logoutAction');
+    await _tryToUploadKeyBackup(matrix: matrix);
+
+    if (PlatformInfos.isMobile) {
+      await _logoutActionsOnMobile(matrix: matrix);
+    } else {
+      await _logoutActionOnOtherPlatform(matrix: matrix);
+    }
+  }
+
+  Future<void> _tryToUploadKeyBackup({required MatrixState matrix}) async {
+    Logs().d('ConnectPageMixin::_tryToUploadKeyBackup');
+    await TwakeDialog.showFutureLoadingDialogFullScreen(
+      future: () async {
+        await matrix.client.encryption?.keyManager.uploadInboundGroupSessions();
+      },
+    );
+  }
+
+  Future<void> _logoutActionsOnMobile({required MatrixState matrix}) async {
+    await tryLogoutSso(matrix: matrix);
```

Comment:
it was removed on the latest change. Please remove it too.

Reply:
Updated